### PR TITLE
Initial Setup

### LINKED ISSUE #1: Basic application setup with Routing
Setup the application with auth features and basic routing

### PR DESCRIPTION
**Pull request by ttahm3d**
Setup basic routes
Header and Sidebar
Footer

Closes issue #2

### CODE REVIEW

File: src/router/Router.jsx [hunk 1]
```diff
@@ -5,6 +5,7 @@ import { Loader } from "../components";
 const Homepage = lazy(() => import("../pages/homepage/Homepage"));
 const Archive = lazy(() => import("../pages/archive/Archive"));
 const Notes = lazy(() => import("../pages/notes/Notes"));
+const Trash = lazy(() => import("../pages/trash/Trash"));
```
Comment (vaibhav-mahalle) [inline src/router/Router.jsx:8]: learnt a new way of importing the pages.

**Review by vaibhav-mahalle — APPROVED** [review]
Learnt the following things while reviewing the pr.

- committing the code feature-wise and in an orderly manner.
- really liked how the commits are ordered starting with just logo and sequentially building up from there.
- learnt a new way or rather a better way of importing pages i.e. lazy loading 
- also learnt about scroll to top feature

File: .env [hunk 2]
```diff
@@ -0,0 +1,2 @@
+REACT_APP_SUPABASE_URL=https://ovyahpfgmosajwzovkmq.supabase.co
```
Comment (srejitk) [inline .env:1]: Why is the Env file with the key being committed?

File: src/hooks/useLocalStorage.jsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      setStoredValue(valueToStore);
+      if (typeof window !== "undefined") {
+        window.localStorage.setItem(key, JSON.stringify(valueToStore));
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  return [storedValue, setValue];
```
Comment (srejitk) [inline src/hooks/useLocalStorage.jsx:26]: Really liked how you've approached localStorage handling and making a component for it.

**Review by srejitk — APPROVED** [review]
Really liked how you've approached breaking down the repeatable aspects into components.
Also Left out one important comment regarding env variables.

**Pull request merged**